sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -93,93 +97,201 @@ public abstract class ServiceBusReceiverClient : IAsyncDisposable
+        //    string connectionString,
+        //    string entityName,
+        //    SessionOptions sessionOptions) :
+        //    this(
+        //        connectionString,
+        //        entityName,
+        //        sessionOptions,
+        //        new ServiceBusReceiverClientOptions())
+        //{
+        //}
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusReceiver"/> class.
         /// </summary>
         ///
         /// <param name="connectionString">The connection string to use for connecting to the Service Bus namespace; it is expected that the shared key properties are contained in this connection string, but not the Service Bus entity name.</param>
         /// <param name="entityName">The name of the specific Service Bus entity to associate the consumer with.</param>
         /// <param name="clientOptions">A set of options to apply when configuring the consumer.</param>
-        /// <param name="sessionId"></param>
+        /// <param name="sessionOptions"></param>
         ///
         /// <remarks>
         ///   If the connection string is copied from the Service Bus entity itself, it will contain the name of the desired Service Bus entity,
         ///   and can be used directly without passing the <paramref name="entityName" />.  The name of the Service Bus entity should be
         ///   passed only once, either as part of the connection string or separately.
         /// </remarks>
         ///
-        internal ServiceBusReceiverClient(
+        internal ServiceBusReceiver(
             string connectionString,
             string entityName,
-            string sessionId,
+            SessionOptions sessionOptions,
             ServiceBusReceiverClientOptions clientOptions)
         {
             Argument.AssertNotNullOrEmpty(connectionString, nameof(connectionString));
             Argument.AssertNotNull(clientOptions, nameof(clientOptions));
 
+            IsSessionReceiver = sessionOptions != null;
             OwnsConnection = true;
             Connection = new ServiceBusConnection(connectionString, entityName, clientOptions.ConnectionOptions);
             RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
             ReceiveMode = clientOptions.ReceiveMode;
-            Consumer = Connection.CreateTransportConsumer(retryPolicy: RetryPolicy, sessionId: sessionId);
+            Consumer = Connection.CreateTransportConsumer(
+                retryPolicy: RetryPolicy,
+                prefetchCount: PrefetchCount,
+                sessionId: sessionOptions?.SessionId,
+                isSessionReceiver: IsSessionReceiver);
+            Session = new ServiceBusSession(
+                Consumer,
+                RetryPolicy,
+                sessionOptions != null);
         }
 
-        /// <summary>
-        ///   Initializes a new instance of the <see cref="ServiceBusReceiverClient"/> class.
+        ///// <summary>
```

Comment:
Was leaving these commented intentional?

Reply:
Yeah those will be used if we go back to making this not be abstract -- also this should be abstract, looks like that was deleted at some point.